lib/index.js

```diff
@@ -355,18 +382,21 @@ Publisher.prototype.publish = function (headers, options) {
 
 /**
  * Sync file in stream with file in the s3 bucket
- * @param {String} prefix
+ * @param {Object} options
+ * @param {String} options.prefix prefix to sync a specific directory
+ * @param {Array} options.prefix list of expressions that match against files that should not be deleted
  *
  * @return {Stream} a transform stream that stream both new files and delete files
  * @api public
  */
 
-Publisher.prototype.sync = function(prefix) {
+Publisher.prototype.sync = function(options) {
```

Comment:
I would support adding a second parameter, so we keep backward compatibility (just as you mentioned).  If we get some more features that need extra parameters we will refactor it to use an option, I just don't want to bump the major version yet.